### DISCUSSION TITLE
GPS sensor of the NAO in the competition

### DISCUSSION THREAD

**Discussion by MJavadZallaghi** [cyberbotics/wrestling#73] [Q&A] [answered]
Dear @omichel 
Hi and I hope you are doing well.
I wanted to use the gps sensor of the NAO robot for its positioning, but I got `NoneType` after using `getDevice()` function.
Here is the code:
```python
self.gps = self.getDevice('gps')
self.gps.enable(self.time_step)
```
I have checked the proto file of the NAO robot `NAO.proto` and I didn't find gps sensor in that file. So, I wanted to check with you the availability of the gps sensor in the competition, as it's part of the standard sensors of the NAO, while it seems currently the sensors is not available.

**Comment by omichel** [comment, marked as answer]
This is normal. The GPS was removed from the NAO robot in the wrestling competition because it is not included in the real NAO robot.
Actually, this is a difficult problem to estimate the position and orientation of the robot on the wrestling stage and using a GPS would be kind of cheating. So, to foster the development of more realistic robot controllers, the participants have to rely on other realistic sensors (camera, accelerometer, gyro, etc.) to make an estimate of their position and orientation on the wrestling stage.

By the way, even if a real robot would be equipped with a GPS device (similar to the ones found in mobile phones), the precision of the position would be so bad (especially indoors) that it would be unusable.

**Comment by MJavadZallaghi** [comment]
Thanks for your explanation.
